It's all about the parentheses!.To access most of the functionality in pandas, you have to call the methods and attributes of DataFrame and Series objects. 

In [1]:
import pandas as pd

In [2]:
movies = pd.read_csv('http://bit.ly/imdbratings')

In [3]:
movies.head()

star_rating                     title content_rating   genre  duration  \
0          9.3  The Shawshank Redemption              R   Crime       142   
1          9.2             The Godfather              R   Crime       175   
2          9.1    The Godfather: Part II              R   Crime       200   
3          9.0           The Dark Knight          PG-13  Action       152   
4          8.9              Pulp Fiction              R   Crime       154   

                                         actors_list  
0  [u'Tim Robbins', u'Morgan Freeman', u'Bob Gunt...  
1    [u'Marlon Brando', u'Al Pacino', u'James Caan']  
2  [u'Al Pacino', u'Robert De Niro', u'Robert Duv...  
3  [u'Christian Bale', u'Heath Ledger', u'Aaron E...  
4  [u'John Travolta', u'Uma Thurman', u'Samuel L....

movies.head(): which returns first 5 rows of dataframe.

In [5]:
movies.describe()

star_rating    duration
count   979.000000  979.000000
mean      7.889785  120.979571
std       0.336069   26.218010
min       7.400000   64.000000
25%       7.600000  102.000000
50%       7.800000  117.000000
75%       8.100000  134.000000
max       9.300000  242.000000

movies.describe():  As long as there is atleast one numeric column in dataframe, it will show you descriptive statistics of all numeric columns, here it is 'star_rating' and 'duration'.

So both head() and describe() both have parentheses.

##### Don't have Parentheses are:

In [6]:
movies.shape

(979, 6)

movies.shape: that's a tuple that tells us 979 rows and 6 columns.

In [8]:
movies.dtypes

star_rating       float64
title              object
content_rating     object
genre              object
duration            int64
actors_list        object
dtype: object

movies.dtypes: which tells the data types of each of the 6 columns. floating point column, integer column and object column is nothing but Strings.

In [9]:
type(movies)

pandas.core.frame.DataFrame

movies is a Dataframe object type. Dataframe has certain methods and attributes, just because its a dataframe. Here "Methods" are head() and describe() with parentheses and "Attributes" are shape and dtypes without parentheses. 
Methods are like action oriented and Attributes are just like descriptions about who you are. 
Methods and Attributes both are applied to Dataframe and Series.

In [ ]:
movies.describe(include='object')

Above shows describe() method not only describe numeric column but also object(String) column. it only describe the column type 'object'

##### Helpfull Note: This is only in iPython/Jupyter Notebook:
Whenever you have Method or function and you want to memorise what the argument are just click anywhere inside the parentheses and hit shift tab once shows brief signature and dock string(pop-up), hit shift tab second time the window gets bigger and if you hit shift tab third time it stays there for longer and if you hit shift tab fourth time it takes up whole half of your screen. Super Handy if you are trying to use a function for your reference. Now it displays like split screen. if you want to exit just hit X.